ckan/config/environment.py

```diff
@@ -373,9 +373,19 @@ def template_loaded(template):
     # Create Jinja2 environment
     cache_dir = config.get('jinja2_cache_dir', None)
     if not cache_dir:
-        cache_dir = tempfile.mkdtemp()
+        global TEMP_JINJA2_CACHE_DIR
```

Comment:
Nooooooo, don't do this, go back to the last version you had which was perfect. You do NOT need or want to clean the cache directory if one is specified, Jinja is doing that for you with bytecode verification.

Globals are *BAD* and should be avoided at all costs.